api/envoy/watchdog/v3alpha/abort_action.proto

```diff
@@ -0,0 +1,29 @@
+syntax = "proto3";
+
+package envoy.watchdog.v3alpha;
+
+import "google/protobuf/duration.proto";
+
+import "udpa/annotations/status.proto";
+import "udpa/annotations/versioning.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.watchdog.v3alpha";
+option java_outer_classname = "AbortActionProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).work_in_progress = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Watchdog Action that kills a stuck thread to kill the process.]
```

Comment:
Did you consider making this a "core extension"? See https://github.com/envoyproxy/envoy/blob/master/source/extensions/all_extensions.bzl#L18. I guess the distinction is whether this is something that Envoy can operate without in principle, (e.g. it can work without `router`) or if it's needed for the basic build and operation of the server.

Reply:
It seems like there's some confusion about what a "core" extension is -- perhaps I'm using the wrong terminology. Essentially from https://github.com/envoyproxy/envoy/pull/13208#discussion_r494454273 the requirement is that the "extension" is always linked.

Unlike the examples in https://github.com/envoyproxy/envoy/blob/master/source/extensions/all_extensions.bzl#L18 the files aren't meant to be in `//source/extensions/`. 

Linking it directly might be the best way to ensure that. If it was clobbered on import, Envoy would likely crash if kill or multikill are on since they'd look for the factory and be unable to find it. Assuming they somehow didn't crash, if they had a watchdog kill event, without this action the process wouldn't actually die which would be bad.

Reply:
Just to close the loop here, I suggested that this get moved back to core. IMO if we assume this functionality must be linked (has to be in the binary) it doesn't make sense to have it be an extension.